msteams-platform/bots/how-to/conversations/conversation-messages.md

```diff
@@ -137,6 +137,117 @@ async def on_message_activity(self, turn_context: TurnContext):
+> * If a tenant admin or user disables the **Read receipts** setting, the bot doesn't receive the read receipt event.
+
+To receive read receipts events for your bot, ensure the following:
+
+* Add the [RSC](~/graph-api/rsc/resource-specific-consent.md#rsc-permissions-for-a-chat-or-meeting) `ChatMessageReadReceipt.Read.Chat` permission in the [app manifest](~/resources/schema/manifest-schema.md), as follows: 
+
+    # [App manifest v1.12 or later](#tab/app-manifest-v112-or-later)
+    
+    ```json
+    
+    "webApplicationInfo": {
+    
+        "id": "38f0ca43-1c38-4c39-8097e-47f62c686500",
+        "resource": ""
+    },
+    "authorization": {
+        "permissions": {
+            "orgwide": [],
+            "resourceSpecific": [
+            {
+                "name": "ChatMessageReadReceipt.Read.Chat",
+                "type": "Application"
+            }
+            ]
+        }
+    }
+    
+    ```
+    
+    # [App manifest v1.11 or earlier](#tab/app-manifest-v111-or-earlier)
+    
+    ```json
+    
+     “webApplicationInfo”: {
+    
+         "id": "123456c8-67d2-4f54-b74e-408b195c4cbc",
+         "resource": "https: //AnyString",
+         "applicationPermissions": [
+             "ChatMessageReadReceipt.Read.Chat"
+         ]
+     },
+    
+    ```
+    
+    ---
+    
+    You can also add RSC permissions through Graph API. For more information, see [`consentedPermissionSet`](/graph/api/userteamwork-teamsappinstallation-upgrade#http-request).
+
+* Override the method `OnTeamsReadReceiptAsync` with `IsMessageRead` handler.
+
+  The `IsMessageRead` helper method is useful to determine if the message is read by the recipients. If the `compareMessageId` is less than or equal to the `LastReadMessageId`, then the message has been read. Override the `OnTeamsReadReceiptAsync` method to receive read receipts with [`IsMessageRead`](/dotnet/api/microsoft.bot.schema.teams.readreceiptinfo.ismessageread#microsoft-bot-schema-teams-readreceiptinfo-ismessageread(system-string)) helper method:
+
+    ```csharp
+    
+    protected override async Task OnTeamsReadReceiptAsync(ReadReceiptInfo readReceiptInfo, ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken) 
+    
+    ```
+
+    Following is an example of read receipts event request that a bot receives:
+    
+    ```json
+    {
+        "name": "application/vnd.microsoft.readReceipt",
+        "type": "event",
+        "timestamp": "2023-08-16T17:23:11.1366686Z",
+        "id": "f:b4783e72-9d7b-2ed9-ccef-ab446c873007",
+        "channelId": "msteams",
+        "serviceUrl": "https://canary.botapi.skype.com/amer/",
+        "from": {
+            "id": "29:1-8Iuh70W9pRqV8tQK8o2nVjxz33RRGDKLf4Bh7gKnrzN8s7e4vCyrFwjkPbTCX_Co8c4aXwWvq3RBLr-WkkVMw",
+            "aadObjectId": "5b649834-7412-4cce-9e69-176e95a394f5"
+        },
+        "conversation": {
+            "conversationType": "personal",
+            "tenantId": "528dbe3f-16e0-4e37-84a2-00cc305847ed",
```

Comment:
tenant id should be 6babcaad-604b-40ac-a9d7-9fd97c0b779f

Reply:
Updated